floris/turbine_library/turbine_previewer.py

```diff
@@ -0,0 +1,905 @@
+import re
+from math import ceil
+from pathlib import Path
+
+import attrs
+import matplotlib.pyplot as plt
+import numpy as np
+from attrs import define, field
+
+from floris.simulation import (
+    Ct,
+    power,
+    Turbine,
+)
+from floris.type_dec import NDArrayFloat
+from floris.utilities import load_yaml
+
+
+INTERNAL_LIBRARY = Path(__file__).parent
+DEFAULT_WIND_SPEEDS = np.linspace(0, 40, 81)
+
+
+def round_nearest_2_or_5(x: int | float) -> int:
+    """Rounds a number (with a 0.5 buffer) up to the nearest integer divisible by 2 or 5.
+
+    Args:
+        x (int | float): The number to be rounded.
+
+    Returns:
+        int: The rounded number.
+    """
+    base_2 = 2
+    base_5 = 5
+    return min(base_2 * ceil((x + 0.5) / base_2), base_5 * ceil((x + 0.5) / base_5))
+
+
+def round_nearest(x: int | float, base: int = 5) -> int:
```

Comment:
Same as above, I wonder if this should go in utilities. I can see this being generally useful.

Reply:
That makes a lot of sense. I wasn't sure if it was too niche of use case, but I guess it's fairly general, and could be used in the future, so I moved both instances.